trie/triedb/pathdb/layertree.go

```diff
@@ -120,7 +121,8 @@ func (tree *layerTree) cap(root common.Hash, layers int) error {
 	}
 	diff, ok := l.(*diffLayer)
 	if !ok {
-		return fmt.Errorf("triedb layer [%#x] is disk layer", root)
+		log.Warn("triedb cap layer is disk layer", "root", root.String())
+		return nil
```

Comment:
What scenarios will be triggered this error?  Please add some comments.

Reply:
The modifications here are easy to understand and I don't think comments are necessary. In some abnormal scenarios, when commit is called but no new difflayer is added into triedb, corresponding errors will occur. For example, it will exit without imporing blocks after startup.

Reply:
```
In some abnormal scenarios, when commit is called but no new difflayer is added into triedb, corresponding errors will occur. For example, it will exit without imporing blocks after startup.
```

Actually, the above are the scenarios and comments I want to know. Why Ethereum returns an error and we just print the warning log?

Reply:
revert